lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs

```diff
@@ -237,6 +244,30 @@ spec = do
                 "Invalid genesis hash: expecting a hex-encoded \
                 \value that is 32 bytes in length"
 
+    describe "SERVER - Clean shutdown" $ do
+        it "SERVER - shuts down on command" $ \ctx -> do
+            logged <- withLogCollection $ \stream -> do
+                let cmd = Command
+                        (commandName @t)
+                        ["serve"
+                        , "--node-port", show (ctx ^. typed @(Port "node"))
+                        , "--random-port"
+                        , "--genesis-block-hash", block0H
+                        ]
+                        (pure ())
+                        CreatePipe
+                        stream
+
+                withBackendProcessHandle nullTracer cmd $ \(Just hStdin) _ -> do
+                    threadDelay oneSecond
+                    hClose hStdin
```

Comment:
**
[question]
***

it all comes down to this when shutting down on cmd, right? No way that we have some kind of shutdown without closing stdin handle?

Reply:
Yes good point - I added a check to make sure that the process exits with status 0.